Source/HTTP/httpcall.cpp

```diff
@@ -371,7 +374,8 @@ void retry_http_call_until_done(
     auto httpSingleton = get_http_singleton(false);
     if (nullptr == httpSingleton)
     {
-        XAsyncComplete(retryContext->outerAsyncBlock, S_OK, 0);
+        HC_TRACE_WARNING(HTTPCLIENT, "Http call after HCCleanup was called. Aborting call.");
```

Comment:
> HC_TRACE_WARNING(HTTPCLIENT, "Http call after HCCleanup was called. Aborting call."); [](start = 7, length = 86)

Does HC_TRACE_WARNING rely on the global state?

Reply:
No, it uses

TraceState& GetTraceState() noexcept
{
    static TraceState state;
    return state;
}

and bails out if it that's not setup